### PR TITLE
Removing the deprecated since 2.2.0 JS files

### LINKED ISSUE #33532: Remove the 2.2.0 deprecated files
### Description

The deprecated since 2.2.0 JS files, that aren't used have to be removed.

### AC

- The deprecated 2.2.0 JS files are removed

### PR DESCRIPTION
**Pull request by eduard13**
### Description (*)
This PR aims to remove the deprecated JS files since 2.2.0.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#33532 - Removing the deprecated files with prototype usages

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by eduard13** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@magento run Functional Tests B2B, Functional Tests CE

File: app/code/Magento/GiftMessage/view/frontend/requirejs-config.js [hunk 1]
```diff
@@ -7,9 +7,7 @@ var config = {
     map: {
         '*': {
             giftOptions:    'Magento_GiftMessage/js/gift-options',
-            extraOptions:   'Magento_GiftMessage/js/extra-options',
-            'Magento_GiftMessage/gift-options':    'Magento_GiftMessage/js/gift-options',
-            'Magento_GiftMessage/extra-options':   'Magento_GiftMessage/js/extra-options'
+            'Magento_GiftMessage/gift-options':    'Magento_GiftMessage/js/gift-options'
```
Comment (andrewbess) [inline app/code/Magento/GiftMessage/view/frontend/requirejs-config.js:10]: Hello @eduard13 
Perhaps this is redundant mapping
Could you please check it?
Thank you in advance.

**Review by andrewbess — CHANGES_REQUESTED** [review]
Hello @eduard13 
Thank you for your contriution.
Could you please check my change request.

File: app/code/Magento/GiftMessage/view/frontend/requirejs-config.js (hunk 1 above)
Comment (eduard13) [reply app/code/Magento/GiftMessage/view/frontend/requirejs-config.js:10]: I agree that is a redundant declaration, however the purpose of this task/PR is another - to get rid of deprecated code.

Comment (andrewbess) [reply app/code/Magento/GiftMessage/view/frontend/requirejs-config.js:10]: Ok. Thanks.
Approved

**Comment by eduard13** [comment]
@magento run Functional Tests B2B

**Comment by andrewbess** [comment]
@magento run Functional Tests B2B

**Comment by andrewbess** [comment]
@magento run Functional Tests B2B

**Comment by andrewbess** [comment]
@magento run Functional Tests B2B

**Review by Den4ik — APPROVED** [review]
✅ Looks like all is ok.
@eduard13 Thanks for the work

**Review by andrewbess — APPROVED** [review]
Hello @eduard13
Thank you for your contribution.
The changes are approved.

**Comment by eduard13** [comment]
Hi @Den4ik, removing something is not always an easy thing, that can be marked as `Cleanup`. First, it should be carefully checked if isn't anywhere used (which is a really time consuming thing, mostly for more than 30 of files), and can be safely removed, then make sure that there is no Jasmine test failing after removing them.

Thanks.

**Comment by sivaschenko** [comment]
@Den4ik I wouldn't mark libraries elimination as cleanup. Such work requires extensive investigation. The "Cleanup" label would suit better for small local changes that do not affect the framework from a global perspective. i.e typos, inline documentation, coding style, remove unused code, and minor style/design fixes.

(https://devdocs.magento.com/contributor-guide/contributing.html#points)

**Comment by Den4ik** [comment]
@eduard13 @sivaschenko I discussed it with @andrewbess and I don't want underestimate @eduard13 work.
I agree that done big work, but I'm try to give an objective assessment and this assessment base only on changes.
DevDocs said:
> Contribution contains code cleanup, such as typos, inline documentation, coding style, **remove unused code**, and minor style/design fixes.

In my opinion it's not fair set "Cleanup" label for removing 3 files and set "special achievement" for removing 30 files.
But it's just mine subjective opinion.

PS. Agree with "category of expertise"

**Pull request merged**